heap/heap.py

```diff
@@ -70,6 +70,10 @@ def _sift_down(self, index):
       # if true swap index and largest and recurse
       self.storage[index], self.storage[largest] = self.storage[largest], self.storage[index]
       self._sift_down(index)
+    # check to see if still a valid max-heap
+    for i in range(self.get_size() - 1):
+      if self.storage[i + 1] > self.storage[i]:
+        self._bubble_up(i + 1)
```

Comment:
FYI, the sift_down method should only need to worry about getting the index that got passed, to the lowest it should be in the BST, and it shouldn't need to perform a bubble_up